pkg/providerconfig/types.go

```diff
@@ -135,13 +135,16 @@ func (cvr *ConfigVarResolver) GetConfigVarBoolValueOrEnv(configVar providerconfi
 	if err != nil {
 		return false, err
 	}
-	if stringVal == "" {
+
+	// if stringVar is "false", it might have been not set and we can attempt to read from environment.
+	// if it is "true", we know it was configured via value or secret reference.
+	if stringVal == "" || stringVal == strconv.FormatBool(false) {
```

Comment:
Shouldn't we use and here? If someone puts `false` in the ProviderSpec, and they have env var set to true, I think we should respect what's in the ProviderSpec. With this `if`, we would ignore `false` value and still try to read from env, which could override it.

Reply:
Problems like this are usually solved with additional bool field which will clearly indicate ether the value has need set explicitly or just empty.
Example: https://pkg.go.dev/database/sql#NullBool

```
type NullBool struct {
	Bool  bool
	Valid bool // Valid is true if Bool is not NULL
}
```

`.Bool` field is the value, and .Valid is indicator if it was set/present/not-null.

We can do the same trick with `ConfigVarBool`, since it's employ custom json unmarshall method anyway.

Reply:
That sounds sensible, I'll look into it. Thanks for the suggestion!

Reply:
Done, please review the new implementation.